src/components/Facility/ConsultationDetails/PrintQuestionnaireQuestionnaireResponses.tsx

```diff
@@ -0,0 +1,381 @@
+              </h1>
+              <h2 className="text-gray-500 uppercase text-sm tracking-wide mt-1 font-semibold">
+                {t("encounter_questionnaire_logs")}
+              </h2>
+            </div>
+          </div>
+
+          <EncounterDetails encounter={encounter} />
+
+          <div className="flex flex-col sm:flex-row justify-between items-center sm:items-start mb-4 pb-2 border-b">
+            <div className="text-center sm:text-left sm:order-1">
+              <h3 className="text-lg font-semibold">{questionnaire?.title}</h3>
+              <p className="text-gray-500 text-sm tracking-wide mt-1">
+                {questionnaire?.description}
+              </p>
+            </div>
+          </div>
+
+          {questionnaireResponses?.results?.map(
+            (item: QuestionnaireResponse) => (
+              <div key={item.id} className="w-full">
+                <ResponseCard key={item.id} item={item} />
+              </div>
+            ),
+          )}
+        </div>
+      </div>
+    </PrintPreview>
+  );
+}
+
+const DetailRow = ({
+  label,
+  value,
+  isStrong = false,
+}: {
+  label: string;
+  value?: string | null;
+  isStrong?: boolean;
+}) => {
+  return (
+    <div className="flex">
+      <span className="text-gray-600 w-32">{label}</span>
+      <span className="text-gray-600">: </span>
+      <span className={`ml-1 ${isStrong ? "font-semibold" : ""}`}>
+        {value || "-"}
+      </span>
+    </div>
+  );
+};
+
+interface EncounterDetailsProps {
+  encounter?: Encounter;
+}
+
+export function EncounterDetails({ encounter }: EncounterDetailsProps) {
+  const { t } = useTranslation();
+
+  if (!encounter) return null;
+
+  return (
+    <div className="grid grid-cols-1 md:grid-cols-2 gap-x-12 gap-y-6 mb-8">
+      <div className="space-y-3">
+        <DetailRow
+          label={t("patient")}
+          value={encounter?.patient.name}
+          isStrong
+        />
+        <DetailRow
+          label={`${t("age")} / ${t("sex")}`}
+          value={
+            encounter?.patient
+              ? `${formatPatientAge(encounter.patient, true)}, ${t(`GENDER__${encounter.patient.gender}`)}`
+              : undefined
+          }
+          isStrong
+        />
+      </div>
+      <div className="space-y-3">
+        <DetailRow
+          label={t("encounter_date")}
+          value={
+            encounter?.period?.start &&
+            format(new Date(encounter.period.start), "dd MMM yyyy, EEEE")
+          }
+          isStrong
+        />
+        <DetailRow
+          label={t("mobile_number")}
+          value={
+            encounter?.patient.phone_number &&
+            formatPhoneNumber(encounter.patient.phone_number)
+          }
+          isStrong
+        />
+      </div>
+    </div>
+  );
+}
+
+function formatValue(value: ResponseValue["value"], type: string): string {
+  if (!value) return "";
+
+  if (
+    typeof value === "object" &&
+    !Array.isArray(value) &&
```

Comment:
Is the not array check needed? Since with the switch statement, seems like it would be a single value.